tests/integration/src/install_tests.rs

```diff
@@ -0,0 +1,1009 @@
+            "expected {} permissions, got {}",
+            expected_permissions.len() + expected_extra_permissions,
+            listed_permissions.permissions.len()
+        ));
+    }
+
+    Ok(())
+}
+
+fn assert_initial_request_policies(
+    env: &PocketIc,
+    canister_id: Principal,
+    requester: Principal,
+    expected_request_policies: &[InitRequestPolicyInput],
+    expected_extra_request_policies: usize,
+) -> Result<(), String> {
+    let listed_request_policies = list_request_policies(env, canister_id, requester)
+        .expect("failed to get request policies")
+        .0
+        .expect("failed to get request policies");
+
+    if listed_request_policies.policies.len()
+        != expected_request_policies.len() + expected_extra_request_policies
+    {
+        return Err(format!(
+            "expected {} request policies, got {}",
+            expected_request_policies.len() + expected_extra_request_policies,
+            listed_request_policies.policies.len()
+        ));
+    }
+
+    Ok(())
+}
+
+fn assert_initial_named_rules(
+    env: &PocketIc,
+    canister_id: Principal,
+    requester: Principal,
+    expected_named_rules: &Vec<InitNamedRuleInput>,
+) -> Result<(), String> {
+    let listed_named_rules = list_named_rules(env, canister_id, requester)
+        .expect("failed to get named rules")
+        .0
+        .expect("failed to get named rules");
+
+    if expected_named_rules.len() != listed_named_rules.named_rules.len() {
+        return Err(format!(
+            "expected {} named rules, got {}",
+            expected_named_rules.len(),
+            listed_named_rules.named_rules.len()
+        ));
+    }
+
+    for expected_named_rule in expected_named_rules {
+        let _named_rule = listed_named_rules
+            .named_rules
+            .iter()
+            .find(|named_rule| named_rule.name == expected_named_rule.name)
+            .ok_or(format!("named rule {} not found", expected_named_rule.name))?;
+    }
+
+    Ok(())
+}
+
+fn assert_initial_assets(
+    env: &PocketIc,
+    canister_id: Principal,
+    requester: Principal,
+    expected_assets: &Vec<InitAssetInput>,
+) -> Result<(), String> {
+    let listed_assets = list_assets(env, canister_id, requester)
+        .expect("failed to call list_assets")
+        .0
+        .expect("failed to list assets");
+
+    if expected_assets.len() != listed_assets.assets.len() {
+        return Err(format!(
+            "expected {} assets, got {}",
+            expected_assets.len(),
+            listed_assets.assets.len()
+        ));
+    }
+
+    for expected_asset in expected_assets {
+        let asset = listed_assets
+            .assets
+            .iter()
+            .find(|asset| asset.name == expected_asset.name)
+            .ok_or(format!("asset {} not found", expected_asset.name))?;
+
+        if expected_asset
+            .id
+            .as_ref()
+            .map(|id| id != &asset.id)
+            .unwrap_or(false)
```

Comment:
Do we want to ignore the case of no expected asset ID?

Reply:
The expected asset has an optional ID, and if it's None, this check should evaluate to false, so it won't result in an error immediately. Am I missing something?

Reply:
> it won't result in an error immediately

My question is if this is desirable: why would the expected asset have its ID equal to None?

Reply:
The expected asset is of type `InitAssetInput`, whose `id` field is optional.